core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -24,6 +24,15 @@
 #include <cstdint>
 #include <string>
 
+// Use likely, unlikely attributes from C++20 to improve performance of
+// reference counting when copy constructors are involved.
+#if defined(KOKKOS_ENABLE_IMPL_REF_COUNT_BRANCH_LIKELY) && \
+    defined(KOKKOS_ENABLE_CXX20)
+#define KOKKOS_BRANCH_PROB [[likely]]
+#else
+#define KOKKOS_BRANCH_PROB [[unlikely]]
+#endif
```

Comment:
This won't work. It should be
```suggestion
#if defined(KOKKOS_ENABLE_CXX20)
#define KOKKOS_BRANCH_PROB [[unlikely]]
#else
#define KOKKOS_BRANCH_PROB
#endif
```
to conform to @crtrott's suggestion.

Reply:
Right. We are only considering adding `[[unlikely]]`.